core/src/main/java/com/linecorp/armeria/client/ClientRequestContext.java

```diff
@@ -372,16 +372,105 @@ ClientRequestContext newDerivedContext(RequestId id, @Nullable HttpRequest req,
+     *
+     * @deprecated Use {@link #adjustResponseTimeoutMillis(long)} or {@link #resetResponseTimeoutMillis(long)}
      */
+    @Deprecated
     void setResponseTimeoutMillis(long responseTimeoutMillis);
 
     /**
      * Sets the amount of time allowed until receiving the {@link Response} completely
      * since the transfer of the {@link Response} started. This value is initially set from
      * {@link ClientOption#RESPONSE_TIMEOUT_MILLIS}.
+     *
+     * <p>For example:
+     * <pre>{@code
+     * ClientRequestContext ctx = ...;
+     * ctx.setResponseTimeoutMillis(1000);
+     * assert ctx.responseTimeoutMillis() == 1000;
+     * // Do something...
+     * ctx.setResponseTimeoutMillis(2000);
+     * assert ctx.responseTimeoutMillis() == 2000;
+     * }</pre>
+     * @deprecated Use {@link #adjustResponseTimeout(Duration)} or {@link #resetResponseTimeout(Duration)}
      */
+    @Deprecated
     void setResponseTimeout(Duration responseTimeout);
 
+    /**
+     * Adjusts the existing response timeout allowed by the specified {@code adjustmentMillis} until
+     * receiving the {@link Response} completely since the transfer of the {@link Response} started.
+     * This value is initially set from {@link ClientOption#RESPONSE_TIMEOUT_MILLIS}.
+     *
+     * <p>For example:
+     * <pre>{@code
+     * ClientRequestContext ctx = ...;
+     * long oldResponseTimeoutMillis = ctx.responseTimeoutMillis();
+     * ctx.adjustResponseTimeoutMillis(1000);
+     * assert ctx.responseTimeoutMillis() == oldResponseTimeoutMillis + 1000;
+     * }</pre>
+     */
+    void adjustResponseTimeoutMillis(long adjustmentMillis);
+
+    /**
+     * Adjusts the existing response timeout allowed by the specified {@code adjustment} until
+     * since the transfer of the {@link Response} started.
+     * This value is initially set from {@link ClientOption#RESPONSE_TIMEOUT_MILLIS}.
+     *
+     * <p>For example:
+     * <pre>{@code
+     * ClientRequestContext ctx = ...;
+     * long oldResponseTimeoutMillis = ctx.responseTimeoutMillis();
+     * ctx.adjustResponseTimeout(Duration.ofSeconds(1));
+     * assert ctx.responseTimeoutMillis() == oldResponseTimeoutMillis + 1000;
+     * }</pre>
+     */
+    void adjustResponseTimeout(Duration adjustment);
+
+    /**
+     * Sets the new response timeout allowed that is after the specified {@code responseTimeoutMillis}
+     * from the now until receiving the {@link Response} completely since the transfer
+     * of the {@link Response} started.
+     * This value is initially set from {@link ClientOption#RESPONSE_TIMEOUT_MILLIS}.
+     *
+     * <p>For example:
+     * <pre>{@code
+     * ClientRequestContext ctx = ...;
+     * // Do something...
+     * ctx.resetResponseTimeoutMillis(1000);
+     * long passedTimeFromStartMillis = ...;
+     * assert ctx.responseTimeoutMillis() == passedTimeFromStartMillis + 1000;
+     * }</pre>
+     */
+    void resetResponseTimeoutMillis(long responseTimeoutMillis);
```

Comment:
Reset isn't clicking for me, it's just setting really. What if it has the word after? `setResponseTimeoutAfter`?

Reply:
It was meant to reset or restart the response timeout from 'now'. 😅 
`setResponseTimeoutAfter` is good to me. But I think the name has ambiguous on the base time.
How about `setResponseTimeoutAfterNow`? it is long though...